### PR TITLE
Add labels to the global styles variations

### LINKED ISSUE #38918: Theme Variant "Other" Styles UI Proposal
Currently, each style variation is "styled" to resemble the actual theme.json variants. 

As I've explored theme variants more, a common issue I'm seeing is that the **style variation components can look quite similar to each other** — especially if the background color hasn't changed. And I'd say it's very likely that this will be the case.

## What we have today

Below is an example of the current styles interface, with four different styles, of which three have very different color palettes, although the background and foreground colors are the same. We don't have labels to name each style, but even if we did, this UI is confusing as three of the four options look exactly the same.

CleanShot 2022-02-18 at 15 59 15@2x

## Proposal: Bringing color palettes into the variant UI

By design, these variants can be as bold and creative as the designer intends (to the point where each variant is completely different), or as simple as a different color palette. You would never know in the current UI pictured above that each of the four styles have very different color palettes to set the tone of the style. 

So let's consider adding a color palette to a style variant's selector (perhaps with a max of five, like we do in the Global Styles > Colors panel) so that more of the actual _style_ of the variant is depicted in the UI.  I've made a few other changes design-wise to accommodate, but the original concept is still the same. 

Those same four styles above, would come across like this: 

CleanShot 2022-02-18 at 16 02 15@2x

And here is a screenshot that would resemble some of the variants explored with Twenty Twenty Two added (to show more simple variants like above, with more stylistic variants like we're exploring in TT2:

CleanShot 2022-02-18 at 16 02 53@2x

## Wrapping up

Overall, having more clarity with style variants will help folks understand what they are changing and applying to their site. 

I have a [quick prototype here](https://www.figma.com/proto/OPJwrJoOGJ6BGCnsfhaLSD/Global-Styles?page-id=0%3A1&node-id=2%3A12&viewport=250%2C48%2C1&scaling=min-zoom&starting-point-node-id=2%3A12) to see it first hand. 

Let me know what you think :)

### PR DESCRIPTION
**Pull request by youknowriad**
## What?

closes #38918

This PR adds "labels" to the global styles variations shown in the site editor sidebar.

## Why?

Something the colors and the previews are not enough to differentiate between the available variations. This PRs allows theme authors to provide a name for each style variation to disambiguate between them.

## How?

Theme authors can add a top level `name` property to the `json` file for each variation. If no name is provided, we fallback to the file name (without the extension).

I'll be needing some help here to make this new property translatable. cc @oandregal @swissspidy 

## Testing Instructions

 - Add a `theme.json` like file (same format as theme.json but different name potentially) to the `styles` folder of your theme
 - Add a top level `name` property in the JSON
 - Open the site editor
 - On the styles sidebar (if it's closed, you should be able to open it by clicking the "Styles" button on the top toolbar), you should be able to click "Other styles".
 - A list of available style variations should be shown corresponding to the alternative theme.json files in the `styles` folder of your theme. Each style variation should have the proper label corresponding to the "name" property in the theme.json file.

## Screenshots or screencast 

https://user-images.githubusercontent.com/272444/157490447-ea009b38-6d85-4bad-a6e2-648d047596ac.mov

### CODE REVIEW

**Comment by jasmussen** [comment]
You're so fast it's hard to keep up!

The initial mockup of labels actually made me think that, their utility aside, the primary challenge with these variations remains that they all look alike. And in that light, they risk actually adding a little noise, and distract from the main problem of diversity in the appearance. I think it could be worth revisiting how the card is composed; in a little longer term I'd love to see everything from font pairings, various border radiuses, maybe even photo elements as part of it. 

And indeed the title of the variation could be a part of the card itself, perhaps on hover, letting the resting state of the card remain a "style stamp". I don't completely love the following fake animation, but just to high level illustrate the idea, here's what a style called "Stratus" could look like when hovering:

Animation

**Comment by critterverse** [comment]
Thanks for trying this out @youknowriad! ⚡

I like @jasmussen's idea of using animation to incorporate the label/swatches — seems that could help resolve both of the main points discussed in https://github.com/WordPress/gutenberg/issues/38918.

**Comment by alexstine** [comment]
@youknowriad Can you add more specific instructions? I need to know what to click and search for to find this section. I'm not sure I can make it there since FSE is still in super rough shape for screen readers but I'd like to give it a try.

Thanks.

**Comment by swissspidy** [comment]
To make this translatable, the i18n schema needs to be updated here and in WP-CLI

**Comment by youknowriad** [comment]
@alexstine I've detailed the testing instructions in the PR description a bit more, let me know if there's anything missing, I'm happy to detail more.

**Comment by youknowriad** [comment]
I've been thinking a bit about @jasmussen's proposal above. I like it a lot but I also fear that it doesn't solve the problem fully. 

Basically, the issue is that unless you hover the preview, you might not notice a difference between the variations. I'll give it a try but I wonder if there's a middle ground where we have both: the hover effect and the visible small label

**Comment by jasmussen** [comment]
> Basically, the issue is that unless you hover the preview, you might not notice a difference between the variations. I'll give it a try but I wonder if there's a middle ground where we have both: the hover effect and the visible small label

This feels like enough of a limitation of the card design itself that it feels worth addressing. Foreground and background colors feel solid enough, but perhaps the two dots show something other than text and link. The label is valuable and should definitely be visible at least on hover. But on its own it's not going to make things "scannable" or inviting, which for style variations is arguably the most important problem to solve.

**Comment by oandregal** [comment]
> I'll be needing some help here to make this new property translatable.

Unfortunately, this is one of those cases in which we need to update the code in addition to the i18n schema.

@swissspidy I've prepared an initial PR for the wp-cli at https://github.com/wp-cli/i18n-command/pull/306 I'll follow up with the required changes for Gutenberg.

**Comment by youknowriad** [comment]
@oandregal before committing these, I want to check with folks whether we want this to be called "name" or "title" or something else :).

**Comment by swissspidy** [comment]
@oandregal Note that you need to update https://github.com/WordPress/gutenberg/blob/f9c2838a0b8be94fcf4e51b0a831d4629eaf24d7/lib/compat/wordpress-5.9/theme-i18n.json as well with the same changes.

> Add a `theme.json` like file (same format as theme.json but different name potentially) to the styles folder of your theme

I see how using `theme.json` for theme _styles_ can be confusing. It would be good to settle on the name first before updating WP-CLI. Because if other file names are supported as well, the string extraction needs to be aware of these file names.

**Comment by oandregal** [comment]
> @oandregal before committing these, I want to check with folks whether we want this to be called "name" or "title" or something else :).

I still need to prepare the tests for the wp-cli, so no worries, we can wait until this PR lands.

I'd also like confirmation about what we want the context to be (what's shown to translators). I went with "Name of the style variation" but thought I'd ask if we envision this field to be used for something else, in which case it could be worth using a more generic approach such as "Name of the style source", or something along those lines.

> It would be good to settle on the name first before updating WP-CLI. Because if other file names are supported as well, the string extraction needs to be aware of these file names.

:+1:

**Comment by alexstine** [comment]
@youknowriad I found the Styles button and it was expanded but that button does not manage focus. Once I arrive in the sidebar, what is the button I am looking for?

**Comment by youknowriad** [comment]
@alexstine Inside the sidebar, the button should be labeled "Other styles" (that's actually a visible label inside it)

**Comment by alexstine** [comment]
@youknowriad Still don't see it. When I make it to the "StylesBeta" sidebar, these are my options.
1. More Global Styles Actions.
2. Close global styles sidebar.
3. Typography.
4. Colors.
5. Layout.
6. Blocks.

Any thoughts?

**Comment by youknowriad** [comment]
@alexstine Do you have a "styles" folder in your theme with some variations. (I'm mentioning this on the testing instructions)

> Add a theme.json like file (same format as theme.json but different name potentially) to the `styles` folder of your theme

File: packages/edit-site/src/components/global-styles/screen-style-variations.js
Comment (ntsekouras) [inline packages/edit-site/src/components/global-styles/screen-style-variations.js]: If we go with this approach, we need to add some width 'limits/handling'.
Screenshot 2022-03-14 at 11 43 58 AM

**Comment by alexstine** [comment]
@youknowriad Okay. I guess I won't see the button until that part is done. I really don't know what I am doing with block themes at this point. Is there any theme already that has this close?

Thanks.

**Comment by ntsekouras** [comment]
>@youknowriad Okay. I guess I won't see the button until that part is done. I really don't know what I am doing with block themes at this point. Is there any theme already that has this close?

@alexstine you can use the `empty theme` provided by GB. It has a variation that doesn't do anything useful in practice, but you can test the style variations UI.

**Comment by alexstine** [comment]
@ntsekouras I enabled the theme. How should I modify this file?

```
bash-5.1$ pwd
/var/www/html/wp-content/themes/emptytheme/styles
bash-5.1$ cat variation.json
{
        "version": 2,
        "settings": {
                "color": {
                        "palette": [
                                {
                                        "slug": "foreground",
                                        "color": "#3F67C6",
                                        "name": "Foreground"
                                }
                        ]
                }
        },
        "styles": {
                "blocks": {
                        "core/post-title": {
                                "typography": {
                                        "fontWeight": "700"
                                }
                        }
                }
        }
}
```

Thanks.

**Comment by ntsekouras** [comment]
>@ntsekouras I enabled the theme. How should I modify this file?

@alexstine just add a top level `name` property with the desired name.
Example:
```
{
	"version": 2,
	"name": "test name",
	.....
        .....
}
```

File: lib/compat/wordpress-6.0/class-gutenberg-rest-global-styles-controller.php
Comment (alexstine) [inline lib/compat/wordpress-6.0/class-gutenberg-rest-global-styles-controller.php:236]: Not sure the name of the file is quite right for using in this case. Maybe mark it as "No name provided"? That would make more sense to me than hearing the name of the file.

File: lib/compat/wordpress-6.0/class-gutenberg-rest-global-styles-controller.php
Comment (alexstine) [inline lib/compat/wordpress-6.0/class-gutenberg-rest-global-styles-controller.php]: Not sure if it is handled anywhere else, but maybe want to make this an empty check instead? That way you can't pass:

```
"name": ""
```

**Review by alexstine — COMMENTED** [review]
@youknowriad Some thoughts.

@ntsekouras Thanks for helping me test this.

File: lib/compat/wordpress-6.0/class-gutenberg-rest-global-styles-controller.php
Comment (youknowriad) [reply lib/compat/wordpress-6.0/class-gutenberg-rest-global-styles-controller.php:236]: The filename in empty theme example is not very representative IMO and I feel like most file names will be in this case personally.

**Comment by youknowriad** [comment]
https://user-images.githubusercontent.com/272444/158345347-8e88114d-ffc0-4305-ab44-b6a8a8942485.mov

@jasmussen I've added some animations to show the first 4 colors of the palette on hover. So far, when I tried animating the label as well, I didn't like the result. Let me know what you think about the current state.

**Comment by jasmussen** [comment]
Nice, that's fun. I think those animations are a bit fast, and the positioning of the dots is a bit close on the edge. The circles also feel not totally round anymore, and the card is a bit tall (stuff we can also look at as part of #38934). 

But mostly this feels very promising in terms of bringing variety to the cards! The particular layout of the dots, what to surface and so on, could probably use a full separate iteration. If we were to do this, could we make the labels appear as part of this animation on focus and hover?

**Comment by youknowriad** [comment]
> If we were to do this, could we make the labels appear as part of this animation on focus and hover?

We can, but I didn't like the result personally, long labels get cut and it felt just too much to me 🤷‍♂️

**Comment by youknowriad** [comment]
I moved the label to the preview animation (I still prefer it outside personally). Anyway, I'd love more clarity on the expected result here so we can land this.

**Comment by jasmussen** [comment]
Just wanted to note that I've seen some sketches @critterverse has shared that look great. She's AFK today, but I suspect she'll share here very soon. Thanks for your patience Riad!

**Comment by critterverse** [comment]
Hey all, @jasmussen and I have been collaborating on some sketches for an animated hover sequence in [this Figma file](https://www.figma.com/file/uJGfIRWh2uwwMdJd5xfLj7/Global-Styles-switcher?node-id=735%3A8362). We’ve explored a few different directions (see Figma for sketches and notes on each), but we wanted to highlight one direction in particular that felt like it was working well so far.

Here are some motion tests for a “color fill” hover animation. The goal of this approach was to try to show the maximum amount of color possible by having swatches from the theme palette take over the entire thumbnail preview. There may even be an opportunity to weight the color bars dynamically based on how much that particular color is used in the theme.

light

dark

preview-hover-animation

The animation could resolve on a frame that contains additional information about the variation.

color-fill-storyboard

Some notes on the third frame:

- The variation title could be displayed on the first line. This line can also double as a preview of Heading text.
- The smaller Aa text underneath could display a preview of paragraph text (left) and link text (right)
- Showing the swatches as a bar/line style (rather than circular) on this frame gives us more room to display the typography styles. This look was inspired from some early iterations on the main Global Styles preview by @pablohoneyhoney

Thoughts?

**Comment by youknowriad** [comment]
- I like the first and last frame
 - I like that the label has a dedicate "row", it gives us some room for long labels but there's still the question of what to do if it exceeds the line length.
 - I feel the second frame is maybe unnecessary and just add a feeling of heaviness to the animation personally.

**Comment by jasmussen** [comment]
Great thoughts, Riad. I like to think that an elaborate hover animation can give weight and value to each card, but I hear you it can feel heavy. We can start with less and see how that feels. A quick riff based on your comments is to just have these two frames, and a quick horizontal wipe between them:

non-weighted

If we embraced the "weighted" concept, i.e. orange is used once in theme.json, olive is used 2, white is used 7, gray is used 8, it could look like this:

weighted

> there's still the question of what to do if it exceeds the line length.

I like that this _encourages_ short and consistent variation names. Less is more, and "Blue" is better than "TwentyTwentyTwo Blue Style Variation". I think it's worth trying up to the point of eliding the text if it's too long. It's very easy to revisit and add tooltips and external labels on top of that.

Speaking of card improvements, I'm seeing this for TwentyTwentyTwo card variations:
Screenshot 2022-03-23 at 08 10 24

All those 6 are different styles, and I understand that this is technically working insofar as the Paragraph font is leveraged, as are background, text, and link colors. These styles all mainly diverge in headings and accent colors, which is what would make the card come alive, though, and I have an idea I wanted to run by you in terms of how we could bring more variety to this.

Here's "Mint", which @kjellr has explored for TwentyTwentyTwo, including an imagined style card:

mint

But the style card looks like this:
Screenshot 2022-03-23 at 08 16 05

This is because the Paragraph font is system fonts, the background is white, and text and links are black. So it's technically working, but it fails a bit in conveying a different visual. I think we can potentially do something smart here. Here's the color palette for the above:

```
			"palette": [
				{
					"slug": "foreground",
					"color": "#000000",
					"name": "Foreground"
				},
				{
					"slug": "background",
					"color": "#FFFFFF",
					"name": "Background"
				},
				{
					"slug": "primary",
					"color": "#000000",
					"name": "Primary"
				},
				{
					"slug": "secondary",
					"color": "#93FFDE",
					"name": "Secondary"
				},
				{
					"slug": "tertiary",
					"color": "#93FFDE",
					"name": "Tertiary"
				}
			]
```

That particular style variation doesn't come with a font family definition of its own, so presumably it falls back to those registered in theme.json:

```
			"fontFamilies": [
				{
					"fontFamily": "-apple-system,BlinkMacSystemFont,\"Segoe UI\",Roboto,Oxygen-Sans,Ubuntu,Cantarell,\"Helvetica Neue\",sans-serif",
					"name": "System Font",
					"slug": "system-font"
				},
				{
					"fontFamily": "\"Source Serif Pro\", serif",
					"name": "Source Serif Pro",
					"slug": "source-serif-pro"
				}
			],
```

So here's the idea, and let me know if it makes sense or is feasible. Instead of building the card from background, text and link colors, as well as the paragraph font, we built it based on the order of the above arrays?

Say we wanted to create a card like this for Mint:

Frame 2

I'd then create the following color and font family arrays:

```
			"palette": [
				{
					"slug": "background",
					"color": "#FFFFFF",
					"name": "Background"
				},
				{
					"slug": "primary",
					"color": "#000000",
					"name": "Primary"
				},
				{
					"slug": "secondary",
					"color": "#93FFDE",
					"name": "Secondary"
				},
				{
					"slug": "tertiary",
					"color": "#93FFDE",
					"name": "Tertiary"
				}
				{
					"slug": "foreground",
					"color": "#000000",
					"name": "Foreground"
				},
			]
```

```
			"fontFamilies": [
				{
					"fontFamily": "\"Bodoni Moda\", serif",
					"name": "Bodoni Moda",
					"slug": "bodoni-moda"
				}
				{
					"fontFamily": "-apple-system,BlinkMacSystemFont,\"Segoe UI\",Roboto,Oxygen-Sans,Ubuntu,Cantarell,\"Helvetica Neue\",sans-serif",
					"name": "System Font",
					"slug": "system-font"
				},
			],
```

Then, simply by applying the first three colors in sequence, and using the first font family registered, it would affect the appearance of the card. What do you think?

**Comment by youknowriad** [comment]
> Then, simply by applying the first three colors in sequence, and using the first font family registered, it would affect the appearance of the card. What do you think?

I'm not sure it's a great idea to base the card on the order of colors... The reason is that we might end up with things that don't make sense (dark color on a dark background for instance) while text/link colors are by default meant to be rendered on top of background color with good contrast.

**Comment by jasmussen** [comment]
The main motivation for using the order was to find a mechanism to add diversity to the cards without adding any new APIs. The current experience of 6 cards all with white backgrounds, dark dots and the same font isn't useful either, and I'd argue it's already the responsibility of the theme designer to choose colors in a way that avoided contrast issues. 

I'm assuming we want to avoid adding an API to let theme developers choose which properties make up the style card, right? What other options do we have that lets us get to the proposed cards? Using the Heading font for the Aa seems a good start. Can we use "primary" and "secondary" for the two dots, if the json file has colors that start with those terms?

**Comment by youknowriad** [comment]
> Aa seems a good start. Can we use "primary" and "secondary" for the two dots, if the json file has colors that start with those terms?

We don't have a "primary" and "secondary" colors, this concept doesn't exit. Indeed, I'd like to avoid a new API as it seems more of a UI choice of the current implementation and something that could evolve over time. I guess this problem was the primary reason for folks to suggest showing the palette in the first frame.

**Comment by ntsekouras** [comment]
Do you think that showing the variations' colors in smaller circles(like we do in palettes) would better hint users about the end result? Even with a new API that explicitly dictates what to show in preview, the actual usage of colors in blocks/root can vary a lot based on the final design.

**Comment by jasmussen** [comment]
> I guess this problem was the primary reason for folks to suggest showing the palette in the first frame.

Right, and that may still be valid. How does this sound:

* Use background and foreground colors and Heading block font for Aa
* Show the first two colors of the palette as the dots.
* Hover shows the expanded palette as mocked up, and with smaller Aa's that show text and link colors and styles, in addition to the smaller style title in heading font above.

Edit: the key reason to stick with a stringent Aa and two dots for the main card is that it provides a uniform neat resting state that is egalitarian between all variations. I'd like to exhaust our options here before we change that.

**Comment by youknowriad** [comment]
@jasmussen I can try it, I think it will result in some of the two dots being invisible though. I guess I can show the first two colors from the palette that are not the same as the background color.

**Comment by jasmussen** [comment]
> I think it will result in some of the two dots being invisible though. I guess I can show the first two colors from the palette that are not the same as the background color.

I think that could work!

**Comment by youknowriad** [comment]
@jasmussen Ok, so I implemented the frames and add a light fade in/fade out animation for now, Let me know what you think. We can also try a slide in animation instead

**Comment by jasmussen** [comment]
This already feels better:
card

I feel like there are still some missing pieces, perhaps to the style variations which [I got from here](https://github.com/WordPress/twentytwentytwo/pull/333/files). But even with style variations that all have a white background (is that intentional @kjellr or do I have old files?) this already increases the diversity.

* I suspect we can't show text-decoration styles until we're further with #27075 is that right?
* Can we also use the font-weight for Aa? Swiss as shown in the GIF above has a bolder heading style which is missing from the card.
* I think the fade works. I think a slide would be even better, but this definitely seems fine to launch for now.
* The titles are they derived from the filename, i.e. `blue.json`? If so can we text-transform to titlecase?
* I'm seeing some issues with tabbing between variations, that's probably in trunk as well, but I have to press tab twice between style variations to focus the next one. Can we play the animation also on focus?

There are some metrics here and there that I can likely dive in and fix, but I think that can work fine separately to this PR so nothing that needs to block it. Which means this one feels close to landing!

**Comment by youknowriad** [comment]
> Can we also use the font-weight for Aa? Swiss as shown in the GIF above has a bolder heading style which is missing from the card.

Sure, for all "Aa"s or just the heading one (main one)

> The titles are they derived from the filename, i.e. blue.json? If so can we text-transform to titlecase?

Ideally each variation defines a "name" property in the json, the filename is just a fallback

> I'm seeing some issues with tabbing between variations, that's probably in trunk as well, but I have to press tab twice between style variations to focus the next one. Can we play the animation also on focus?

I'll try

**Comment by jasmussen** [comment]
> Sure, for all "Aa"s or just the heading one (main one)

For the main card one, and for the title shown on the hover.

> Ideally each variation defines a "name" property in the json, the filename is just a fallback

Ah, perfect!

**Comment by youknowriad** [comment]
@jasmussen I think I addressed all the points.

**Comment by jasmussen** [comment]
This is already looking way better to me:
Screenshot 2022-03-23 at 14 19 49

Focus works great here 👌 — so well in fact, I feel it validates the hover idea.

The duplicated greens and pinks in the attached is because the variation duplicates the same color, so probably something to fix there. 

I'd like to dive in and tweak some of the metrics of the second frame, but I need a break from the keyboard. Would you prefer I do this in a followup PR separately? I'd be happy to, and I'd also be happy to try this one as is, pending any thoughts from @kjellr and @critterverse.

**Comment by youknowriad** [comment]
> I'd like to dive in and tweak some of the metrics of the second frame, but I need a break from the keyboard. Would you prefer I do this in a followup PR separately?

I think a follow-up would be great mainly to unblock some related work in i18n. cc @oandregal 

I'm still looking for confirmation that "name" is the best property name for the label, I don't know, "title" or "label" could be valid alternatives. cc @mtias @mcsf

**Comment by kjellr** [comment]
Here's the current state of the Twenty Twenty-Two variations (https://github.com/WordPress/wordpress-develop/pull/2440), tested against this PR: 

styles

In general (and aside from the fonts not working yet... if someone has a sec to help look into that, I'd appreciate it!), it feels pretty good! The mint + pink color variants are showing the same circle swatch color twice, but they do define it twice at the moment — I'm probably going to go in and adjust that on the theme side anyway.

**Comment by critterverse** [comment]
Awesome, this is working so much better! Thanks @youknowriad ❤️

>I'd like to dive in and tweak some of the metrics of the second frame

This sounds great, @jasmussen. I'd also like to follow up with a simplified animation suggestion for transitioning between the two frames that could be addressed separately as well.

**Comment by youknowriad** [comment]
@kjellr I can also filter duplicates in colors.

**Comment by youknowriad** [comment]
@kjellr Actually, I just tried filtering duplicates and I ended up with just a unique color dot for both these variants which felt a bit weird as a result. So maybe duplicates are fine for now.

**Comment by kjellr** [comment]
> Actually, I just tried filtering duplicates and I ended up with just a unique color dot for both these variants which felt a bit weird as a result. So maybe duplicates are fine for now.

That's fine, we can always re-address later.

**Comment by critterverse** [comment]
Hey all, I shared a revised version of the [color fill animation approach](https://github.com/WordPress/gutenberg/pull/39322#issuecomment-1075251944) in https://github.com/WordPress/gutenberg/issues/39700. It includes the sliding color bars but reduces the complexity of the hover frame to only show the variation title as Heading text, which I think should help with the animation feeling too weighty. 

dark

**Review by ntsekouras — APPROVED** [review]
Looks good, thanks Riad! Let's land and iterate.

**Comment by oandregal** [comment]
Status on translating the `title` key: the wp-cli i18n-command PR is ready https://github.com/wp-cli/i18n-command/pull/306 but blocked by these PRs that update the i18n schema:

- Gutenberg https://github.com/WordPress/gutenberg/pull/39936
- WordPress `trunk` https://github.com/WordPress/wordpress-develop/pull/2483

**Comment by beckej13820** [comment]
I really like the mockups where I can see the name without hovering. 

I also really like the mockups that clearly show the majority of colors in the pallette, because I am imaging themes might want to have a variation of style in several colors. This example is a customizer theme, but Go had multiple font/style choices, and for each had multiple color palletes.
Screen Shot 2022-04-26 at 2 40 10 PM

If they were recreating the same sort of flexibility it might be there are 20 variations

Trendy Plum, Trendy Steel, Trendy Avocado, Trendy Champagne
Welcoming ______, Welcoming _____....

**Comment by critterverse** [comment]
>If they were recreating the same sort of flexibility it might be there are 20 variations

@beckej13820 Not exactly the same idea but I shared a design in https://github.com/WordPress/gutenberg/issues/29388#issuecomment-1089507879 recently that would introduce more flexibility for style variations by offering granular Global Styles saving options.

**Pull request merged**
